pkg/flow/flow_components.go

```diff
@@ -32,11 +39,19 @@ func (f *Flow) GetComponent(id component.ID, opts component.InfoOptions) (*compo
 }
 
 // ListComponents implements [component.Provider].
-func (f *Flow) ListComponents(opts component.InfoOptions) []*component.Info {
+func (f *Flow) ListComponents(moduleID string, opts component.InfoOptions) ([]*component.Info, error) {
```

Comment:
IMO I think we need types/doc around global, module, component IDs. Its far to easy to get confused about each one. This may not be the PR for it but when I see this here I feel I have to think about what type of ID it wants.